drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,822 @@
+	struct espi_xec_data *data =
+		(struct espi_xec_data *) (dev->driver_data);
+
+	data->plt_rst_asserted = 0;
+
+	/* Configure eSPI_PLTRST# to cause nSIO_RESET reset */
+	PCR_REGS->PWR_RST_CTRL = MCHP_PCR_PR_CTRL_USE_ESPI_PLTRST;
+	ESPI_CAP_REGS->PLTRST_SRC = MCHP_ESPI_PLTRST_SRC_IS_VW;
+
+	/* Configure the channels and its capabilities based on build config */
+	ESPI_CAP_REGS->GLB_CAP0 |= MCHP_ESPI_GBL_CAP0_VW_SUPP;
+	ESPI_CAP_REGS->GLB_CAP0 |= MCHP_ESPI_GBL_CAP0_PC_SUPP;
+
+	/* Max VW count is 12 pairs */
+	ESPI_CAP_REGS->VW_CAP = 11;
+	ESPI_CAP_REGS->PC_CAP |= MCHP_ESPI_PC_CAP_MAX_PLD_SZ_64;
+
+#ifdef CONFIG_ESPI_CHANNEL_OOB
+	ESPI_CAP_REGS->GLB_CAP0 |= MCHP_ESPI_GBL_CAP0_OOB_SUPP;
+	ESPI_CAP_REGS->OOB_CAP |= MCHP_ESPI_OOB_CAP_MAX_PLD_SZ_73;
+	ESPI_OOB_REGS->TX_ADDR_MSW = 0x1FFFul;
+	ESPI_OOB_REGS->RX_ADDR_MSW = 0x1FFFul;
+	ESPI_OOB_REGS->TX_ADDR_LSW = 0xFFFFul;
+	ESPI_OOB_REGS->RX_ADDR_LSW = 0xFFFFul;
+	ESPI_OOB_REGS->RX_LEN = 0x007F0000;
+#else
+	ESPI_CAP_REGS->GLB_CAP0 &= ~MCHP_ESPI_GBL_CAP0_OOB_SUPP;
+#endif
+
+#ifdef CONFIG_ESPI_CHANNEL_FLASH
+	ESPI_CAP_REGS->GLB_CAP0 |= MCHP_ESPI_GBL_CAP0_FC_SUPP;
+	ESPI_CAP_REGS->GLB_CAP0 |= MCHP_ESPI_FC_CAP_MAX_PLD_SZ_64;
+	ESPI_CAP_REGS->FC_CAP |= MCHP_ESPI_FC_CAP_SHARE_MAF_SAF;
+	ESPI_CAP_REGS->FC_CAP |= MCHP_ESPI_FC_CAP_MAX_RD_SZ_64;
+#else
+	ESPI_CAP_REGS->GLB_CAP0 &= ~MCHP_ESPI_GBL_CAP0_FC_SUPP;
+#endif
+
+	/* Clear reset interrupt status and enable interrupts*/
+	ESPI_CAP_REGS->ERST_STS |= MCHP_ESPI_RST_ISTS;
+	ESPI_CAP_REGS->ERST_IEN |= MCHP_ESPI_RST_IEN;
+	ESPI_PC_REGS->PC_STATUS |= MCHP_ESPI_PC_STS_EN_CHG;
+	ESPI_PC_REGS->PC_IEN |= MCHP_ESPI_PC_IEN_EN_CHG;
+
+	/* Enable VWires interrupts*/
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW00,
+		SRC_ID0, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW00,
+		SRC_ID1, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW00,
+		SRC_ID2, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW01,
+		SRC_ID0, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW01,
+		SRC_ID1, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW01,
+		SRC_ID2, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW02,
+		SRC_ID0, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW03,
+		SRC_ID0, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW03,
+		SRC_ID1, MSVW_IRQ_SEL_EDGE_BOTH);
+	mec_espi_msvw_irq_sel_set(&ESPI_M2S_VW_REGS->MSVW03,
+		SRC_ID3, MSVW_IRQ_SEL_EDGE_BOTH);
+
+	/* Enable logic channel enable interrupts */
+	MCHP_GIRQ_ENSET(config->bus_girq_id) = MCHP_ESPI_ESPI_RST_GIRQ_VAL |
+		MCHP_ESPI_VW_EN_GIRQ_VAL | MCHP_ESPI_PC_GIRQ_VAL;
+
+	/* Enable VWires aggregated block interrupts */
+	MCHP_GIRQ_ENSET(config->vw_girq_id) = BIT(0) | BIT(1) | BIT(2) |
+		BIT(5) | BIT(8) | BIT(12);
+
+	MCHP_GIRQ_ENSET(config->pc_girq_id) = BIT(0) | BIT(1);
```

Comment:
If this is GIRQ15 it means you are enabling UART0 and UART1. Remember we moved to uart2 which should be BIT(4). Also add a comment indicating this is a different block

Reply:
all these BIT() call should be hidden behind macros

Reply:
These are virtual peripherals exposed by the eSPI endpoint, there is no direct relation to SoC ones. These ISRs will trigger when eSPI master attempts to access the respective peripheral in the eSPI endpoint and the eSPI controller HW detects that traffic in the bus.